### ISSUE TITLE
ServicePoint breaks REST/SOAP API Schema

### ISSUE THREAD

**Issue by DaimPiek** [creative-ict/sendcloud#43]
**Prerequisites:** 
Magento 2.3.X
Sendcloud 1.0.14

When accessing the following page:
https://hostname.tld/fr/rest/all/schema?services=all

The following stacktrace is shown:

```
#0 /var/www/vendor/magento/module-webapi/Model/Config/ClassReflector.php(103): Magento\Framework\Reflection\TypeProcessor->register('array') 
#1 /var/www/vendor/magento/module-webapi/Model/Config/ClassReflector.php(72): Magento\Webapi\Model\Config\ClassReflector->extractMethodData(Object(Zend\Code\Reflection\MethodReflection)) 
#2 /var/www/vendor/magento/module-webapi/Model/ServiceMetadata.php(134): Magento\Webapi\Model\Config\ClassReflector->reflectClassMethods('SendCloud\\SendC...', Array) 
#3 /var/www/vendor/magento/module-webapi/Model/ServiceMetadata.php(166): Magento\Webapi\Model\ServiceMetadata->initServicesMetadata() 
#4 /var/www/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Webapi\Model\ServiceMetadata->getServicesConfig() 
#5 /var/www/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Webapi\Model\ServiceMetadata\Interceptor->___callParent('getServicesConf...', Array) 
#6 /var/www/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Webapi\Model\ServiceMetadata\Interceptor->Magento\Framework\Interception\{closure}() 
#7 /var/www/generated/code/Magento/Webapi/Model/ServiceMetadata/Interceptor.php(26): Magento\Webapi\Model\ServiceMetadata\Interceptor->___callPlugins('getServicesConf...', Array, Array) 
#8 /var/www/vendor/magento/module-webapi/Model/Rest/Swagger/Generator.php(923): Magento\Webapi\Model\ServiceMetadata\Interceptor->getServicesConfig() 
#9 /var/www/vendor/magento/module-webapi/Controller/Rest/SchemaRequestProcessor.php(53): Magento\Webapi\Model\Rest\Swagger\Generator->getListOfServices() 
#10 /var/www/vendor/magento/module-webapi/Controller/Rest.php(188): Magento\Webapi\Controller\Rest\SchemaRequestProcessor->process(Object(Magento\Framework\Webapi\Rest\Request\Proxy)) 
#11 /var/www/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http)) 
#12 /var/www/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Webapi\Controller\Rest\Interceptor->___callParent('dispatch', Array) 
#13 /var/www/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Webapi\Controller\Rest\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http)) 
#14 /var/www/generated/code/Magento/Webapi/Controller/Rest/Interceptor.php(26): Magento\Webapi\Controller\Rest\Interceptor->___callPlugins('dispatch', Array, Array) 
#15 /var/www/vendor/magento/framework/App/Http.php(137): Magento\Webapi\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http)) 
#16 /var/www/generated/code/Magento/Framework/App/Http/Interceptor.php(24): Magento\Framework\App\Http->launch() 
#17 /var/www/vendor/magento/framework/App/Bootstrap.php(261): Magento\Framework\App\Http\Interceptor->launch() 
#18 /var/www/pub/index.php(40): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor)) 
#19 {main}
```

Issue is the return type of the following methods:
\SendCloud\SendCloud\Model\ServicePoint::activate
\SendCloud\SendCloud\Model\ServicePoint::deactivate
\SendCloud\SendCloud\Model\ServicePoint::shippingEmail

Which shouldn't return an array, instead opt for a DataObject with the necessary getters and setters (for retrieving the messages).

**Comment by aderik** [comment]
Did you also test 1.0.15?

**Comment by DaimPiek** [comment]
The issue seems to be resolved in 1.0.15.
